### ISSUE TITLE
AVT 1 - NumberInput DAP violation that accessible name does not match the visible label.

### ISSUE THREAD

**Issue by snidersd** [carbon-design-system/carbon#4499]
## Environment
macOS 10.14.6
Chrome Version 77.0.3865.90
VoiceOver
DAP 1.8.0.0 Sept 2019 WCAG 2.1 Ruleset

## Description
Run DAP on the React Number Input Component. The following violation is found. (see screenshot):
<img width="1084" alt="Screen Shot 2019-10-30 at 2 33 47 PM" src="https://user-images.githubusercontent.com/21676914/67892502-d6a35900-fb22-11e9-81dc-57546d0cc79a.png">

Note: The issue is that the label is "Number Input label", but the aria-label="Number input field with increment and decrement buttons". If the aria-label is updated to match the actual label the violation will no longer be present.

**Comment by snidersd** [comment]
Also see comments in closed issue #4198.

**Comment by jendowns** [comment]
So if I'm understanding this correctly, the label text should always match the `aria-label` value. And in this case, they are different because both values are populated by separated props.

~In that case, I can propose a quick PR that would apply the `label` prop value (which already populates the label element) to the `aria-label` attribute, and then deprecate the separate `ariaLabel` prop.~

EDIT:
Nevermind, I don't think my approach resolved the issue completely at all, so I closed out my PR 👍

**Comment by dakahn** [comment]
duplicate of #4532
